src/Symfony/Component/HttpKernel/EventListener/ErrorListener.php

```diff
@@ -65,18 +76,25 @@ public function onKernelException(ExceptionEvent $event)
 
             $prev = $e;
             do {
-                if ($exception === $wrapper = $prev) {
+                if ($throwable === $wrapper = $prev) {
                     throw $e;
                 }
             } while ($prev = $wrapper->getPrevious());
 
             $prev = new \ReflectionProperty($wrapper instanceof \Exception ? \Exception::class : \Error::class, 'previous');
             $prev->setAccessible(true);
-            $prev->setValue($wrapper, $exception);
+            $prev->setValue($wrapper, $throwable);
 
             throw $e;
         }
 
+        foreach ($this->exceptionsMapping  as $exception => $config) {
+            if ($throwable instanceof $exception && $config['status_code']) {
+                $response->setStatusCode($config['status_code']);
+                break;
```

Comment:
This means that the order you define the exceptions in your config is important. I don't mind, but this should be documented.